assets/js/components/Work/Tabs/Preservation/FileSetModal.jsx

```diff
@@ -48,179 +45,155 @@ function FileSetModal({ closeModal, isHidden, workId }) {
           "is-success",
           `FileSet record id: ${ingestFileSet.id} created successfully and ${ingestFileSet.metadata.original_filename} was submitted to the ingest pipeline.`
         );
-        handleFormReset();
+        resetForm();
+        closeModal();
       },
       onError(error) {
+        // bug with this error not clearing/resetting
+        // https://github.com/apollographql/apollo-feature-requests/issues/170
         console.error("GQLerror", error);
       },
-      refetchQueries(mutationResult) {
-        return [
-          {
-            query: GET_WORK,
-            variables: { id: workId },
-          },
-        ];
-      },
+      refetchQueries: [
+        {
+          query: GET_WORK,
+          variables: { id: workId },
+        },
+      ],
+      awaitRefetchQueries: true,
     }
   );
 
-  const handleFormReset = () => {
+  const handleSubmit = (data) => {
+    console.log("data", data);
+    ingestFileSet({
+      variables: {
+        accession_number: data.accessionNumber,
+        workId,
+        role: data.role,
+        metadata: {
+          description: data.description,
+          label: data.label,
+          original_filename: currentFile.name,
+          location: s3UploadLocation,
+        },
+      },
+    });
+  };
+
+  const handleCancel = () => {
+    stateXhr.abort();
+    resetForm();
+    closeModal();
+  };
+
+  const resetForm = () => {
     methods.reset();
     setCurrentFile(null);
-    closeModal();
-    clearErrors();
+    setS3UploadLocation(null);
+    setUploadProgress(0);
+    setUploadError(null);
   };
 
   const handleSetFile = (file) => {
     setCurrentFile(file);
+    if (file != null) {
+      uploadFile(file);
+    } else {
+      setUploadProgress(0);
+    }
   };
 
-  const clearErrors = () => {
-    setFileUploadError("");
-  };
+  const uploadFile = (file) => {
```

Comment:
This looks great.  I think in the future we could abstract this out into its own component, or a custom hook... `useFileUploader` or something.